### PR TITLE
[oneDNN] GRU BF16 kernel 

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
New features

### PR changes
OPs

### Describe
Introducing GRU BF16 (brain float 16) kernel implemented via oneDNN library.

### CODE REVIEW

**Comment by jczaja** [comment]
@grygielski Could you please inspect fusion_gru_mkldnn_op.cc changes ?

File: python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+from paddle.fluid.tests.unittests.op_test import OpTest
+from paddle.fluid.tests.unittests.test_fusion_gru_op import fusion_gru
+from paddle.fluid.tests.unittests.test_fusion_lstm_op import fc, ACTIVATION
+
+
+@unittest.skipIf(not core.supports_bfloat16(),
+                 "place does not support BF16 evaluation")
+class TestFusionGRUBF16MKLDNNOp(OpTest):
+    def set_confs(self):
+        self.mkldnn_data_type = False
```
Comment (wozna) [inline python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py:31]: The `mkldnn_data_type` attribute is most commonly used in cpu_bfloat16 passes. In the GRU kernel it is not used, but the type of this attribute is std::string. So you can set `self.mkldnn_data_type = "bfloat16"` or remove this assignment.

Comment (wozna) [reply python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py:31]: But can be done later.

**Review by wozna — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 Could you please start your review? Your approval is needed to pass CI.

**Review by arlesniak — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py (hunk 1 above)
Comment (jczaja) [reply python/paddle/fluid/tests/unittests/mkldnn/test_fusion_gru_bf16_mkldnn_op.py:31]: ok, will do in next PR

**Comment by lidanqing-vv** [comment]
Hi, @luotao1 This PR has only APPROVAL CI failing, please help approve. Thanks

**Review by luotao1 — APPROVED** [review]
LGTM
```
0. Unittest is not allowed to be disabled.
2020-10-01 20:54:05 You must have one RD (kolinwei(Recommend), or luotao1) approval for the usage of @unittest.skip or @unittest.skipIf.
```
You can add a new approval for the usage of `@unittest.skipIf(not core.supports_bfloat16()` in `tools/check_file_diff_approvals.sh` and skip the original one. The one RD (jczaja, xxx, xxx, or luotao1).

Or just skip the approval for the usage of @unittest.skip or @unittest.skipIf when `not core.supports_bfloat16()`

**Pull request merged**